core/src/main/java/org/jruby/util/io/ChannelFD.java

```diff
@@ -131,7 +130,7 @@ private void initChannelTypes() {
         else chRead = null;
         if (ch instanceof WritableByteChannel) chWrite = (WritableByteChannel)ch;
         else chWrite = null;
-        if (ch instanceof SeekableByteChannel) chSeek = (SeekableByteChannel)ch;
```

Comment:
Ugh. I'd really rather not use FileChannel everywhere, even though I don't know of any SeekableByteChannel offhand (even jnr-enxio channels don't implement the interface). And changing chSeek to be a FileChannel makes it essentially just a duplicate of chFile.

I hate to make more work here but this is a pretty icky change to have to make. I wish Google's incorporation of Java 7 APIs weren't so arbitrary.